### PR TITLE
PR for SRVLOGIC-165: Add a tech preview note and change the title to Technology and Developer Preview releases

### PR DESCRIPTION
**Pull request by kaldesai**
Tracking JIRA: https://issues.redhat.com/browse/SRVLOGIC-165

Doc preview: 
Please check the following sections: 

- [Overview](https://deploy-preview-99--jazzy-shortbread-5f62b7.netlify.app/docs/latest/) 
- [Setup Eventing with OpenShift Service Mesh](https://deploy-preview-99--jazzy-shortbread-5f62b7.netlify.app/docs/latest/serverless-eventing/service-mesh/eventing-service-mesh-setup) (Added a DP note)
- [About buildpacks for OpenShift Serverless Functions](https://deploy-preview-99--jazzy-shortbread-5f62b7.netlify.app/docs/latest/functions/serverless-functions-about) (Added a DP note)
- [About OpenShift Serverless Logic](https://deploy-preview-99--jazzy-shortbread-5f62b7.netlify.app/docs/latest/serverless-logic/about) (Added a TP note)

The overview conveys that this URL covers information that is DP and TP and the respective section mentions that they are TP or DP respectively

### CODE REVIEW

File: modules/ROOT/pages/index.adoc [hunk 1]
```diff
@@ -2,15 +2,19 @@
...
 
-Developer Preview releases are not production-ready, and customers are recommended to avoid using the project for production or business-critical workloads.
+Technology and Developer Preview releases are not production-ready, and customers are recommended to avoid using the project for production or business-critical workloads.
+
+OpenShift Serverless provides information about additional, advanced use cases and integrations that are possible using OpenShift Container Platform and OpenShift Serverless, but are for technology and development purposes and are not officially supported by Red Hat.
+
+OpenShift Serverless is a Technology Preview feature only. Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete. Red Hat does not recommend using them in production. These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process.
```
Comment (Preeticp) [inline modules/ROOT/pages/index.adoc:11]: 1. @kaldesai please remove the paragraph from line 11. The rest of the content looks good from the over view pov.
2. In https://deploy-preview-99--jazzy-shortbread-5f62b7.netlify.app/docs/latest/serverless-eventing/service-mesh/eventing-service-mesh-setup, and https://deploy-preview-99--jazzy-shortbread-5f62b7.netlify.app/docs/latest/functions/serverless-functions-about sections, please add notes which provide Developer Preview related disclaimer:
OpenShift Serverless - Developer Preview releases contain features and functionalities that might not be fully tested. Customers are encouraged to provide feedback on Developer Preview releases.
Developer Preview releases are not production-ready, and customers are recommended to avoid using the project for production or business-critical workloads.
3. In https://deploy-preview-99--jazzy-shortbread-5f62b7.netlify.app/docs/latest/serverless-logic/about please add the Technology preview related note:
'Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete. Red Hat does not recommend using them in production.
These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process.'
Basically, the overview conveys that this site covers info that is DP and TP  and the respective section mention that they are TP or DP respectively.

Comment (kaldesai) [reply modules/ROOT/pages/index.adoc]: I have incorporated all of your changes.

**Comment by Preeticp** [comment]
LGTM. I think this is ready to be merged.

**Review by ricardozanini — APPROVED** [review]

**Review by krisv — APPROVED** [review]

**Comment by kaldesai** [comment]
There were conflicts in this PR related to Serverless + Service Mesh integration. So I created a new PR to track all the updated changes Closing this one. 
PR: https://github.com/openshift-knative/docs/pull/101
